src/inference/rejection.js

```diff
@@ -33,28 +40,51 @@ module.exports = function(env) {
     this.oldCoroutine = env.coroutine;
     env.coroutine = this;
 
-    if (!_.isNumber(this.numSamples) || this.numSamples <= 0) {
-      throw new Error('samples should be a positive integer.');
-    }
-
     if (this.incremental && this.maxScore > 0) {
       util.warn('Rejection: Reduce maxScore to zero for better performance.');
     }
   }
 
   Rejection.prototype.run = function() {
+    if (!_.isNumber(this.numSamples) || this.numSamples <= 0) {
+      return this.error('"samples" should be a positive integer.');
+    }
+    var elapseSec = (Date.now() - this.startTime) / 1000.0;
+    if (this.minSampleRate > 0 && elapseSec > 2) {
+      // count the number of samples collected in ~2 secs
+      // compute number of samples per sec
+      var sampleRate = (this.numSamplesTotal - this.numSamples) / 2;
```

Comment:
Even if you only run this check once, it's probably still a good idea to divide by `elapseSec` here (instead of `2`), so that what you compute is closer to "samples per sec", especially if the first runs for some reason take substantially longer than two seconds.